chain/rosetta-rpc/src/lib.rs

```diff
@@ -384,11 +398,54 @@ async fn account_balance(
     } else {
         None
     };
-    Ok(Json(models::AccountBalanceResponse {
-        block_identifier: models::BlockIdentifier::new(block_height, &block_hash),
-        balances: vec![models::Amount::from_yoctonear(balance)],
-        metadata: nonces,
-    }))
+    match currencies {
+        Some(currencies) => {
```

Comment:
You could reduce indentation by 1 level if you use the `if let Some(currencies) = currencies { ... } else { ... }` syntax.